test/e2e/vsphere_volume_placement.go

```diff
@@ -62,319 +80,268 @@ var _ = framework.KubeDescribe("Volume Placement [Feature:Volume]", func() {
+		deletePodAndWaitForVolumeToDetach(c, ns, vsp, node1Name, pod, volumePaths)
+
+		By("Creating pod on the node: " + node1Name + " with volume :" + volumePaths[0] + " and volume:" + volumePaths[1])
+		pod = createPodWithVolumeAndNodeSelector(c, ns, vsp, node1Name, node1KeyValueLabel, volumePaths)
+		// Create empty files on the mounted volumes on the pod to verify volume is writable
+		// Verify newly and previously created files present on the volume mounted on the pod
+		newEmptyFileNames := []string{
+			fmt.Sprintf("/mnt/volume1/%v_2.txt", ns),
+			fmt.Sprintf("/mnt/volume2/%v_2.txt", ns),
+		}
+		volumeFiles = append(volumeFiles, newEmptyFileNames[0])
+		volumeFiles = append(volumeFiles, newEmptyFileNames[1])
+		createAndVerifyFilesOnVolume(ns, pod.Name, newEmptyFileNames, volumeFiles)
+		deletePodAndWaitForVolumeToDetach(c, ns, vsp, node1Name, pod, volumePaths)
+	})
+
+	/*
+		Test Back-to-back pod creation/deletion with different volume sources on the same worker node
+		    1. Create volumes - vmdk2
+		    2. Create pod Spec - pod-SpecA with volume path of vmdk1 and NodeSelector set to label assigned to node1.
+		    3. Create pod Spec - pod-SpecB with volume path of vmdk2 and NodeSelector set to label assigned to node1.
+		    4. Create pod-A using pod-SpecA and wait for pod to become ready.
+		    5. Create pod-B using pod-SpecB and wait for POD to become ready.
+		    6. Verify volumes are attached to the node.
+		    7. Create empty file on the volume to make sure volume is accessible. (Perform this step on pod-A and pod-B)
+		    8. Verify file created in step 5 is present on the volume. (perform this step on pod-A and pod-B)
+		    9. Delete pod-A and pod-B
+		    10. Repeatedly (5 times) perform step 4 to 9 and verify associated volume's content is matching.
+		    11. Wait for vmdk1 and vmdk2 to be detached from node.
+	*/
+	It("test back to back pod creation and deletion with different volume sources on the same worker node", func() {
+		var (
+			podA                *v1.Pod
+			podB                *v1.Pod
+			testvolumePathsPodA []string
+			testvolumePathsPodB []string
+			podAFiles           []string
+			podBFiles           []string
+		)
+
+		defer func() {
+			By("clean up undeleted pods")
+			deletePodWithWait(f, c, podA)
+			deletePodWithWait(f, c, podB)
+			By("wait for volumes to be detached from the node: " + node1Name)
+			for _, volumePath := range volumePaths {
+				waitForVSphereDiskToDetach(vsp, volumePath, types.NodeName(node1Name))
+			}
+		}()
+
+		testvolumePathsPodA = append(testvolumePathsPodA, volumePaths[0])
+
+		// Create another VMDK Volume
+		By("creating another vmdk")
+		volumePath, err := createVSphereVolume(vsp, nil)
+		Expect(err).NotTo(HaveOccurred())
+		volumePaths = append(volumePaths, volumePath)
+		testvolumePathsPodB = append(testvolumePathsPodA, volumePath)
+
+		for index := 0; index < 5; index++ {
+			By("Creating pod-A on the node: " + node1Name + " with volume:" + testvolumePathsPodA[0])
```

Comment:
I believe the prefered pattern is to use `fmt.Sprintf` rather than string contactenation. I belive a reason for this is due to extra intermediate allocations go uses for string addition.